### ISSUE TITLE
option --ignored should not output directories

### ISSUE THREAD

**Issue by mimmoz81** [AlDanial/cloc#196]
I think that running cloc-1.72 with --ignored option to output to a file the ignored files, should not output the ignored folders with "zero sized file" cause.
Having many-many-many folders in a project makes the file nearly unreadable..

![immagine](https://user-images.githubusercontent.com/14213760/26924115-ab79e9ae-4c44-11e7-8c6f-61e8229198e1.png)

**Comment by AlDanial** [comment]
That's curious.   On my file system (ext4), directories do not appear in the ignored list:
<pre>
/tmp> mkdir -p a/b/c
/tmp> echo "date" > a/b/c/d.sh
/tmp> cloc --ignored ign.txt a
       1 text file.
       1 unique file.                              
Wrote ign.txt
       0 files ignored.

github.com/AlDanial/cloc v 1.73  T=0.01 s (97.1 files/s, 97.1 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Bourne Shell                     1              0              0              1
-------------------------------------------------------------------------------
/tmp> cat ign.txt 

/tmp> 
</pre>
The directory ``b`` has no files in it so it should have shown up as having zero size but on my system it doesn't.  Can you try the experiment above on your file system?

**Comment by AlDanial** [comment]
Also, please post the output of these commands when invoked in the same directory as where you launched cloc:
<pre>
perl -e 'printf("%d\n", (stat ".")[2])'
perl -e 'printf("%d\n", -d ".")'
cloc --show-os
</pre>
On my computer the outputs are 16893, 1, and UNIX.

**Comment by mimmoz81** [comment]
I'm running on Windows 10, and FS is NTFS.

```
D:\tmp>mkdir a\b\c

D:\tmp>echo "test" > a\b\c\d.sh

D:\tmp>d:\Util\cloc-1.72.exe --ignored ign.txt a
       1 text file.
       1 unique file.
Wrote ign.txt
       3 files ignored.

github.com/AlDanial/cloc v 1.72  T=0.50 s (2.0 files/s, 2.0 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Bourne Shell                     1              0              0              1
-------------------------------------------------------------------------------

D:\tmp>type ign.txt
a: zero sized file
a/b: zero sized file
a/b/c: zero sized file

D:\tmp>
```

your perl command don't work on my machine.. sorry I'm not using perl since 10 years.. can't help you much with these errors..

```
D:\tmp>perl -e 'printf("%d\n", (stat ".")[2])'
Can't find string terminator "'" anywhere before EOF at -e line 1.

D:\tmp>perl -e 'printf("%d\n", -d ".")'

Loading DB routines from perl5db.pl version 1.49_05
Editor support available.

Enter h or 'h h' for help, or 'perldoc perldebug' for more help.

Can't find string terminator "'" anywhere before EOF at -e line 1.
 at -e line 1.

D:\tmp>

```

**Comment by AlDanial** [comment]
ah, Windows--in that case, for the Perl one-liners, replace all the single quotes with double quotes and vice versa.  I'm almost certain I know what the problem is and will be able to give you a fix easily--that is, if you're running the Perl source code of cloc instead of the exe.

**Comment by mimmoz81** [comment]
```
D:\tmp>perl -e "printf('%d\n', (stat '.')[2])"
16895\n
D:\tmp>perl -e "printf('%d\n', -d '.')"
1\n
D:\tmp>
```

**Comment by AlDanial** [comment]
The fix will be simple.  Is there a chance you can run a Perl interpreter on your machine (eg Strawberry Perl or ActiveState Perl)?  If yes, I'll commit the fix soon, otherwise if you need to wait for the next cloc exe, I plan a release around the end of July.

**Comment by AlDanial** [comment]
Fixed with commit d846f2a
